File: Object-C/BlackFireIM/Example/Lib/MSIMSDK-UIKit/ChatDetail/Model/MSEmotionMessageCellData.m
//
//  BFCustomMessageCellData.m
//  BlackFireIM
//
//  Created by benny wang on 2021/4/13.
//

#import "MSEmotionMessageCellData.h"
#import "MSHeader.h"

@implementation MSEmotionMessageCellData

- (instancetype)init
{
    self = [super init];
    if (self) {
    }
    return self;
}

- (MSIMEmotionElem *)emotionElem
{
    return (MSIMEmotionElem *)self.elem;
}

- (CGSize)contentSize
{
   return TEmotionMessageCell_Container_Size;
}

- (CGFloat)heightOfWidth:(CGFloat)width
{
    CGFloat height = 0;
    if (self.showName) {
        height += 25;
    }
    CGSize containerSize = [self contentSize];
    height += containerSize.height;
    if (self.direction == MsgDirectionOutgoing) {
        height += 20;
    }
    height += 5 + 5;
    return height;
}

- (NSString *)reuseId
{
    return TEmotionMessageCell_ReuseId;
}

@end

File: Swift/MSIM-Swift/Lib/MSIMSDK-UIKit/ChatDetail/Model/MSEmotionMessageCellData.swift
//
//  MSEmotionMessageCellData.swift
//  MSIM-Swift
//
//  Created by benny wang on 2022/2/13.
//

import UIKit
import MSIMSDK


open class MSEmotionMessageCellData: MSMessageCellData {

    public override func contentSize() -> CGSize {
        return MSMcros.TEmotionMessageCell_Container_Size
    }
    
    public override func heightOfWidth(width: CGFloat) -> CGFloat {
        var height: CGFloat = 0
        if self.showName {
            height += 25
        }
        let containerSize = self.contentSize()
        height += containerSize.height
        if self.direction == .outGoing {
            height += 20
        }
        height += 5 + 5
        return height
    }
    
    public override var reUseId: String {
        return MSMcros.TEmotionMessageCell_ReuseId
    }
}
